### PR TITLE
Add named binary authorization policy support for cloud run v2 resources

### PR DESCRIPTION
**Pull request by yanweiguo**
**Release Note Template for Downstream PRs (will be copied)**

```release-note:enhancement
cloudrunv2: added field `binary_authorization.policy` to resource `google_cloud_run_v2_job` and resource `google_cloud_run_v2_service` to support named binary authorization policy.
```

### CODE REVIEW

File: mmv1/third_party/terraform/services/cloudrunv2/resource_cloud_run_v2_job_test.go.erb [hunk 1]
```diff
@@ -213,6 +213,7 @@ func TestAccCloudRunV2Job_cloudrunv2JobWithDirectVPCUpdate(t *testing.T) {
 	jobName := fmt.Sprintf("tf-test-cloudrun-service%s", acctest.RandString(t, 10))
 	context := map[string]interface{}{
 		"job_name": jobName,
+		"project": envvar.GetTestProjectFromEnv(),
```
[the remark this answers is not in the document]
Comment (yanweiguo) [reply mmv1/third_party/terraform/services/cloudrunv2/resource_cloud_run_v2_job_test.go.erb:217]: done

**Comment by yanweiguo** [comment]
BTW, this is the field referred to `policy` in https://cloud.google.com/run/docs/reference/rest/v2/BinaryAuthorization

File: mmv1/third_party/terraform/services/cloudrunv2/resource_cloud_run_v2_job_test.go.erb [hunk 2]
```diff
@@ -270,12 +271,15 @@ func testAccCloudRunV2Job_cloudrunv2JobWithDirectVPC(context map[string]interfac
...
 
-func testAccCloudRunV2Job_cloudrunv2JobWithDirectVPCUpdate(context map[string]interface{}) string {
+func testAccCloudRunV2Job_cloudrunv2JobWithDirectVPCAndNamedBinAuthPolicyUpdate(context map[string]interface{}) string {
 	return acctest.Nprintf(`
   resource "google_cloud_run_v2_job" "default" {
     name     = "%{job_name}"
     location = "us-central1"
-    launch_stage = "BETA"
+    binary_authorization {
+      policy = "projects/%{project}/platforms/cloudRun/policies/test-policy"
```
Comment (BBBmau) [inline mmv1/third_party/terraform/services/cloudrunv2/resource_cloud_run_v2_job_test.go.erb:280]: policy is in the wrong format, it looks like it shouldn't include `/policies/` based on the error output below:

```
        Error: Error updating Job "projects/ci-test-project-188019/locations/us-central1/jobs/tf-test-cloudrun-serviceawtp88lvds": googleapi: Error 400: Violation in UpdateJobRequest.job.binary_authorization.policy: must be in form projects/{project}/platforms/cloudRun/{policy-name}.
        Details:
        [
          {
            "@type": "type.googleapis.com/google.rpc.BadRequest",
            "fieldViolations": [
              {
                "description": "must be in form projects/{project}/platforms/cloudRun/{policy-name}.",
                "field": "Violation in UpdateJobRequest.job.binary_authorization.policy"
              }
            ]
          }
        ]
```

File: mmv1/third_party/terraform/services/cloudrunv2/resource_cloud_run_v2_service_test.go.erb [hunk 3]
```diff
@@ -748,12 +750,16 @@ resource "google_cloud_run_v2_service" "default" {
...
 
-func testAccCloudRunV2Service_cloudRunServiceWithDirectVPCUpdate(context map[string]interface{}) string {
+func testAccCloudRunV2Service_cloudRunServiceWithDirectVPCAndNamedBinAuthPolicyUpdate(context map[string]interface{}) string {
 	return acctest.Nprintf(`
 resource "google_cloud_run_v2_service" "default" {
   name     = "%{service_name}"
   location = "us-central1"
   launch_stage = "GA"
+  binary_authorization {
+    policy = "projects/%{project}/platforms/cloudRun/policies/test-policy"
```
Comment (BBBmau) [inline mmv1/third_party/terraform/services/cloudrunv2/resource_cloud_run_v2_service_test.go.erb:760]: same error output:
```
        Error: Error updating Service "projects/ci-test-project-188019/locations/us-central1/services/tf-test-cloudrun-servicejd0nuj3jis": googleapi: Error 400: Violation in UpdateServiceRequest.service.binary_authorization.policy: must be in form projects/{project}/platforms/cloudRun/{policy-name}.
        Details:
        [
          {
            "@type": "type.googleapis.com/google.rpc.BadRequest",
            "fieldViolations": [
              {
                "description": "must be in form projects/{project}/platforms/cloudRun/{policy-name}.",
                "field": "Violation in UpdateServiceRequest.service.binary_authorization.policy"
              }
            ]
          }
        ]
```

**Comment by yanweiguo** [comment]
It turned out a bug in API side. I'm holding this PR until we fix the issue.

**Comment by yanweiguo** [comment]
@BBBmau could you re-run the tests?

**Review by BBBmau — APPROVED** [review]
Looks great, thanks for this!

**Pull request merged**